src/surface_conditions/surface_conditions.jl

```diff
@@ -381,11 +381,11 @@ end
 #For non-RCEMIPII box models with prescribed surface temp, assume that the latitude is 0.
 function surface_temperature(
     ::Union{ZonallySymmetricSST, ZonallyAsymmetricSST},
-    coordinates::Union{Geometry.XZPoint, Geometry.XYZPoint},
+    coordinates::Union{Geometry.XZPoint, Geometry.XYZPoint, Geometry.ZPoint},
```

Comment:
```suggestion
    coordinates,
```
Unless the coordinate type restriction is being used for dispatch, it should probably be removed (both here and in the other methods for `surface_temperature`).

There's no need for ClimaAtmos to keep track of all point types that include `z`-coordinates. That kind of code should be limited to only ClimaCore's `Geometry` module, where the point types are defined. Mirroring the `Geometry` module's code duplication in ClimaAtmos will only introduce additional friction when we're trying to refactor it later.